io-backend/src/main/java/agh/io/iobackend/controller/GameRoomController.java

```diff
@@ -75,56 +85,96 @@ public ResponseEntity<GameRoomResponse> getRoomDetails(@PathVariable Long id) th
+        }
         gameRoom.setGameStarted(true);
+        Game game = new Game(id, gameRoom.getGameMap());
+        Game savedGame = gameService.createGame(game);
+        gameRoom.setGame(savedGame);
 
-        // szkic pozniejszych zmian z nowa klasa Game podobna do GameRoom
-//        Game game = new Game(id, gameRoom.getMapID(), gameRoom.getGameMasterID());
-//        Game savedGame = gameService.createGame(game);
-//        return ResponseEntity.ok(savedGame.getGameId());
-
-        return ResponseEntity.ok(id);
-
+        return ResponseEntity.ok(savedGame.getGameId());
     }
 
     @CrossOrigin
     @DeleteMapping("/{id}") // room id
-    public ResponseEntity<String> deleteRoom(@PathVariable Long id) throws GameRoomNotFoundException {
-        gameRoomService.deleteGameRoom(id);
-        gameService.deleteGame(id);
+    public ResponseEntity<String> deleteRoom(@PathVariable Long id) {
+        try {
+            gameRoomService.deleteGameRoom(id);
+        } catch (GameRoomNotFoundException e) {
+            return ResponseEntity.badRequest().body("Game room not found");
+        }
+        // cos jeszcze?
         return ResponseEntity.ok("Room deleted");
     }
 
     @CrossOrigin
     @GetMapping("/{id}/users-list") // room id
-    public ResponseEntity<List<User>> getUserListInRoom(@PathVariable Long id) throws GameRoomNotFoundException {
-        GameRoom gameRoom = gameRoomService.getGameRoom(id);
+    public ResponseEntity<List<User>> getUserListInRoom(@PathVariable Long id) {
+        GameRoom gameRoom = null;
+        try {
+            gameRoom = gameRoomService.getGameRoom(id);
+        } catch (GameRoomNotFoundException e) {
+            return ResponseEntity.badRequest().body(null);
+        }
         return ResponseEntity.ok(new ArrayList<>(gameRoom.getUserList()));
     }
 
     @CrossOrigin
     @DeleteMapping("/{id}/users-list/{user}") // room id
-    public void leaveGameRoom(@PathVariable Long id, @PathVariable Long user) throws GameRoomNotFoundException {
+    public void leaveGameRoom(@PathVariable Long id, @PathVariable Long user) throws GameRoomNotFoundException, NoGameFoundException {
         GameRoom gameRoom = gameRoomService.getGameRoom(id);
         gameRoom.removePlayer(userService.getUserById(user).get());
+        if (gameRoom.getUserList().size() == 0) {
+            gameRoomService.deleteGameRoom(id);
+        }
+        gameService.removeFromGame(id, user); //TODO przetestowac
     }
 
     @CrossOrigin
     @PostMapping("/{id}/users-list/{user}") // room id
-    public ResponseEntity<User> joinGameRoom(@PathVariable Long id, @PathVariable Long user) throws GameRoomNotFoundException {
-        GameRoom gameRoom = gameRoomService.getGameRoom(id);
-        User user1 = userService.getUserById(user).get();
-        gameRoom.addPlayer(user1);
-        return ResponseEntity.ok(user1);
+    public ResponseEntity<String> joinGameRoom(@PathVariable Long id, @PathVariable Long user) {
```

Comment:
Zminienił się typ zwracanej wartości, trzeba sprawdzić czy nie wpłynie to na kod na frontendzie